api/v1alpha1/storageconsumer_types.go

```diff
@@ -38,6 +38,7 @@ const (
 )
 
 // StorageConsumerSpec defines the desired state of StorageConsumer
+// +kubebuilder:validation:XValidation:rule="!(has(self.storageQuotaInGiB) && has(oldSelf.storageQuotaInGiB) && self.storageQuotaInGiB < oldSelf.storageQuotaInGiB && self.storageQuotaInGiB != 0)",message="storageQuotaInGiB cannot be decreased unless setting to 0"
```

Comment:
Can you pls explain why we are using not condition and `self.storageQuotaInGiB < oldSelf.storageQuotaInGiB`?

Reply:
We use the ! (not) condition to block updates that try to decrease the storage quota, unless it's being set to 0.